src/components/Calculator.js

```diff
@@ -1,36 +1,47 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
   }
 
+  onClickHandler = (e) => {
+    const result = calculate(this.state, e.target.value);
+    this.setState(result);
+  };
+
   render() {
+    const { total, next } = this.state;
     return (
       <div id="calculator-placeholder" className="calculator-placeholder">
         <h1>Tito&apos; Calculator</h1>
         <div id="calculator" className="calculator">
-          <input type="text" name="input" className="btn input" id="input" />
-          <input type="button" className="btn" value="AC" />
-          <input type="button" className="btn" value="+/-" />
-          <input type="button" className="btn" value="%" />
-          <input type="button" className="btn operator" value="÷" />
-          <input type="button" className="btn" value="7" />
-          <input type="button" className="btn" value="8" />
-          <input type="button" className="btn" value="9" />
-          <input type="button" className="btn operator" value="x" />
-          <input type="button" className="btn" value="4" />
-          <input type="button" className="btn" value="5" />
-          <input type="button" className="btn" value="6" />
-          <input type="button" className="btn operator" value="-" />
-          <input type="button" className="btn" value="1" />
-          <input type="button" className="btn" value="2" />
-          <input type="button" className="btn" value="3" />
-          <input type="button" className="btn operator" value="+" />
-          <input type="button" className="btn" value="0" />
-          <input type="button" className="btn" value="." />
-          <input type="button" className="btn operator" value="=" />
+          <input type="text" name="input" className="btn input" id="input" value={next || total} />
+          <input onClick={this.onClickHandler} type="button" className="btn" value="AC" />
```

Comment:
- [ ] Good job so far with your project implementation it is great :+1:  But I can't perform **"AC"** operation I'm getting errors :cry: , Please fix it in order to make your calculator works as expected :smile: 

- [ ] One more thing I'm getting the error in the console  **"You provided a value prop to a form field without an onChange handler" this error occurs generally when we set a value prop on a field that has no onChange handler.** I suggest you use the defaultValue prop or set an onChange prop on the field. Please fix it to make your project completed. :+1:

Reply:
@codecaiine  Thanks a lot for your time!